src/librustc_mir/transform/const_prop.rs

```diff
@@ -35,6 +35,9 @@ use crate::interpret::{
 use crate::const_eval::error_to_const_error;
 use crate::transform::{MirPass, MirSource};
 
+/// The maximum number of bytes that we'll allocate space for a return value.
+const MAX_ALLOC_LIMIT: u64 = 1024;
```

Comment:
This is perhaps overly conservative. Maybe 1mb would be a better limit?

Reply:
This feels like an overridable value. Like recursion_limit - whatever value you choose someone willl have a good reason for wanting a different value.

Reply:
If you want configurable limits, make an issue for it for some other day. :)

Reply:
recursion_limit affects what code the compiler will accept. With this value, the compiler will still accept your code, it just may not optimize it as heavily as it could. I don't think we should make this configurable without a strong reason.